### PR TITLE
added endpoint to updated team's points

### PR DESCRIPTION
**Pull request by mchlcui**
👷 **Changes:**
I added an endpoint for updating the teams' points field through increments and decrements. 

💭 **Notes:**
It would potentially be beneficial to have an endpoint to set the points field, putting it on hold for the time sake of DataVerse. 

image

**Wait! Before you merge, have you checked the following:**
- [ ] PR is has approving review(s)

### CODE REVIEW

File: services/teams/handler.js [hunk 1]
```diff
@@ -21,6 +21,52 @@ import db from "../../lib/db.js";
...
+
+    const eventIDYear = `${data.eventID};${data.year}`;
+
+    const team = await teamHelpers._getTeamFromUserRegistration(
+      data.user_id,
+      data.eventID,
+      data.year
+    );
+
+    console.log(team);
```
Comment (AllanT102) [inline services/teams/handler.js:43]: remove log

**Comment by AllanT102** [comment]
how are we handling multiple submissions? ie. if someoen tries to resubmit then it shouldn't increment points

**Comment by mchlcui** [comment]
> how are we handling multiple submissions? ie. if someoen tries to resubmit then it shouldn't increment points

This is in progress, I will be adding a field in the TEAMS table that includes a set of the questions that has been answered, so a team cannot be rewarded extra points

**Comment by AllanT102** [comment]
> > how are we handling multiple submissions? ie. if someoen tries to resubmit then it shouldn't increment points
> 
> This is in progress, I will be adding a field in the TEAMS table that includes a set of the questions that has been answered, so a team cannot be rewarded extra points

Instead of introducing a new field in the teams table, are we able to reuse the scannedQrs field or qrscsanhistory field in the registrations table? I forget what the field is called. 

If that's too much work then just create new field then

**Comment by mchlcui** [comment]
fixed, now have endpoints that uses the scannedQrs field to store correctly answered questions. also added an endpoint so questions can be added at once during a quiz submission instead of making an API call for each question answered correctly.

File: services/teams/helpers.js [hunk 2]
```diff
@@ -217,6 +217,54 @@ export default {
...
+        team.scannedQRs.push(...uniqueQuestions);
+
+        const totalPoints = pointsPerQuestion * uniqueQuestions.length;
+
+        if (totalPoints !== 0) {
+          team.points += totalPoints;
+        }
+
+        if (totalPoints < 0) {
+          team.pointsSpent += Math.abs(totalPoints);
```
Comment (AllanT102) [inline services/teams/helpers.js:252]: what is pointSpent?

**Review by AllanT102 — APPROVED** [review]

File: services/teams/serverless.yml [hunk 3]
```diff
@@ -55,6 +55,27 @@ provider:
...
 functions:
+  updateTeamPoints:
+    handler: handler.updateTeamPoints
+    events:
+      - http:
+          path: team/points
+          method: put
+          cors: true
+  addQRScan:
+    handler: handler.addQRScan
```
[the remark this answers is not in the document]
Comment (mchlcui) [reply services/teams/serverless.yml]: no thats a function in the handler, a deprecated call. I removed it though since i wont be using it

**Pull request merged**